ra_aid/__main__.py

```
@@ -285,27 +292,49 @@ def main():
-                )
+        # Validate model configuration early
+        from ra_aid.models_params import models_params
+        model_config = models_params.get(args.provider, {}).get(args.model or "", {})
+        supports_temperature = model_config.get("supports_temperature", args.provider in ["anthropic", "openai", "openrouter", "openai-compatible", "deepseek"])
+        
+        if supports_temperature and args.temperature is None:
+            args.temperature = model_config.get("default_temperature")
+            if args.temperature is None:
+                print_error(f"Temperature must be provided for model {args.model} which supports temperature")
+                sys.exit(1)
+            logger.debug(f"Using default temperature {args.temperature} for model {args.model}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)